### ISSUE TITLE
restore docker container failure: can't open file, no such file or directory

### ISSUE THREAD

**Issue by LiuChangFreeman** [checkpoint-restore/criu#1478]
The whole log is here: [restore.log](https://github.com/checkpoint-restore/criu/files/6494373/restore.log)  
The critical error is:  
```
(00.068476)      1: Error (criu/files-reg.c:1750): Can't open file data/db/mongod.lock on restore: No such file or directory
(00.068482)      1: Error (criu/files-reg.c:1684): Can't open file data/db/mongod.lock: No such file or directory
(00.068485)      1: Error (criu/files.c:1196): Unable to open fd=12 id=0x3b
(00.082797) mnt: Switching to new ns to clean ghosts
(00.090927) Error (criu/cr-restore.c:2293): Restoring FAILED.
```
I'm trying to checkpoint and restore a live `mongodb` container before in which you have helped me a lot https://github.com/checkpoint-restore/criu/issues/1476  
https://github.com/checkpoint-restore/criu/issues/1474  And it's really so kind of all of you to give me suggestions!  

Now this time I want to checkpoint the container in these steps:  

1. Run a `mongodb` container.  
```
docker run -p 127.0.0.1:27017:27017 --name mongo --restart=always --privileged=true  -d mongo:latest 
```
Then insert a lot of rows of data to init the database and make some queries to let it load caches into memory.  
After these operations I get the data persisted in `/data/db` (In the container's own namespace) and a `mongod` process which costs 4096*12%=491M memory.   
Now the `mongod` process opens some files to log and make queries.  

2. Create a checkpoint using the docker criu backend.  
```
docker checkpoint create --checkpoint-dir=/home/checkpoint mongo v1
```
3. Commit the container to make a docker image.  
```
docker commit -p mongo mongo_init:v1
```
4. Create another `mongodb` docker container using the new image. 
```
docker create -p 127.0.0.1:27017:27017 --privileged=true --name mongo_another mongo_init:v1
```
5. Start the docker container using the checkpoint created in step 3.  
```
docker start --checkpoint-dir=/home/checkpoint --checkpoint=v1 mongo_another
```
Finally I got errors at the beginning and it seems that the restorer could't find the files needed by the `mongd` process.  
I'm wondering how to avoid this error without mounting the container's `/data/db` to host machine.  
Many Thanks!

**Comment by adrianreber** [comment]
The interesting question is 'why does `docker commit` not save the missing file `data/db/mongod.lock`'. It could be that `/data` is a Volume and Volumes are not part of the checkpoint.

Just recently @rst0git added Volume support to Podman's checkpoint restore implementation. Maybe you could try to use Podman to run your checkpoint/restore tests and have more luck.

**Comment by LiuChangFreeman** [comment]
> The interesting question is 'why does `docker commit` not save the missing file `data/db/mongod.lock`'. It could be that `/data` is a Volume and Volumes are not part of the checkpoint.
> 
> Just recently @rst0git added Volume support to Podman's checkpoint restore implementation. Maybe you could try to use Podman to run your checkpoint/restore tests and have more luck.

Thank you. I also tried to add files to the `/data` Volume manaully by using `docker cp` after step 4. But it still didn't work. I'll try to use podman later.

**Comment by LiuChangFreeman** [comment]
These steps run well on `podman` so the failure was caused by docker.
